src/skimage/registration/_affine.py

```
@@ -0,0 +1,1106 @@
+            try:
+                # Solve the system
+                r = np.linalg.solve(h_mat.T @ g_mat @ h_mat, h_mat.T @ b)
+
+                # update the matrix
+                matrix = matrix @ self._parameter_vector_to_matrix(r.ravel(), ndim)
+
+                if np.linalg.norm(r) < self._tol:
+                    break
+
+            except np.linalg.LinAlgError:
+                warnings.warn("Failed to invert linear system (np.linalg.solve).")
```

Ruff (0.15.0)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)